app/javascript/dashboard/composables/useAI.js

```diff
@@ -0,0 +1,201 @@
+    getters['draftMessages/get'].value(draftKey.value)
+  );
+
+  /**
+   * Fetches integrations if they haven't been loaded yet.
+   * @returns {Promise<void>}
+   */
+  const fetchIntegrationsIfRequired = async () => {
+    if (!appIntegrations.value.length) {
+      await store.dispatch('integrations/get');
+    }
+  };
+
+  /**
+   * Records analytics for AI-related events.
+   * @param {string} type - The type of event.
+   * @param {Object} payload - Additional data for the event.
+   * @returns {Promise<void>}
+   */
+  const recordAnalytics = async (type, payload) => {
+    const event = OPEN_AI_EVENTS[type.toUpperCase()];
+    if (event) {
+      track(event, {
+        type,
+        ...payload,
+      });
+    }
+  };
+
+  /**
+   * Cleans and normalizes a list of labels.
+   * @param {string} labels - A comma-separated string of labels.
+   * @returns {string[]} An array of cleaned and unique labels.
+   */
+  const cleanLabels = labels => {
+    return labels
+      .toLowerCase() // Set it to lowercase
+      .split(',') // split the string into an array
+      .filter(label => label.trim()) // remove any empty strings
+      .map(label => label.trim()) // trim the words
+      .filter((label, index, self) => self.indexOf(label) === index);
+  };
+
+  /**
+   * Fetches label suggestions for the current conversation.
+   * @returns {Promise<string[]>} An array of suggested labels.
+   */
+  const fetchLabelSuggestions = async () => {
+    if (!conversationId.value) return [];
+
+    try {
+      const result = await OpenAPI.processEvent({
+        type: 'label_suggestion',
+        hookId: hookId.value,
+        conversationId: conversationId.value,
+      });
+
+      const {
+        data: { message: labels },
+      } = result;
+
+      return cleanLabels(labels);
+    } catch (error) {
+      return [];
+    }
+  };
+
+  /**
+   * Processes an AI event, such as rephrasing content.
+   * @param {string} [type='rephrase'] - The type of AI event to process.
+   * @returns {Promise<string>} The generated message or an empty string if an error occurs.
+   */
+  const processEvent = async (type = 'rephrase') => {
+    try {
+      const result = await OpenAPI.processEvent({
+        hookId: hookId.value,
+        type,
+        content: draftMessage.value,
+        conversationId: conversationId.value,
+      });
+      const {
+        data: { message: generatedMessage },
+      } = result;
+      return generatedMessage;
+    } catch (error) {
+      useAlert(t('INTEGRATION_SETTINGS.OPEN_AI.GENERATE_ERROR'));
+      return '';
+    }
+  };
+
+  onMounted(() => {
+    fetchIntegrationsIfRequired();
+  });
+
+  return {
+    draftMessage,
+    uiFlags,
+    appIntegrations,
+    currentChat,
+    replyMode,
+    isAIIntegrationEnabled,
+    isLabelSuggestionFeatureEnabled,
+    isFetchingAppIntegrations,
+    fetchIntegrationsIfRequired,
+    recordAnalytics,
+    fetchLabelSuggestions,
+    processEvent,
+    cleanLabels,
```

Comment:
Do we need to export `cleanLabels`? Also, we can move it outside the composable if not

Reply:
@scmmishra I exported it to write specs for the `cleanLabels`. That's the only case. Otherwise, it's only using in `useAI` composable.

Reply:
@scmmishra Should I need to move it to a helper file?

Reply:
Yeah, lets move it outside

Reply:
@iamsivin we can move it outside the composable, within the file also, I'll push the changes